File: C#/Programming Basics with C#/12. Nested Loops/06. Travelling/Travelling.cs
using System;

class Travelling
{
    static void Main()
    {
        string destination = " ";
        double tripPrice = 0;
        double sum = 0;

        for (destination = Console.ReadLine(); destination != "End"; destination = Console.ReadLine())
        {
            tripPrice = double.Parse(Console.ReadLine());
            while (tripPrice > sum)
            {
                sum = sum + double.Parse(Console.ReadLine());
            }
            Console.WriteLine($"Going to {destination}!");
            sum = 0;
        }
    }
}

File: C++/Programming Basics with C++/12. Nested loops/simple_tasks/Travelling.cpp
#include<iostream>
#include <string>

using namespace std;

int main() {

    string destination;
    int tripPrice = 0;
    int sum = 0;

    for (getline(cin, destination); destination != "End"; getline(cin, destination)) {
        string input;
        getline(cin, input);
        tripPrice = stoi(input);

        while (tripPrice > sum) {
            getline(cin, input);
            sum += stoi(input);
        }

        cout << "Going to " << destination << "!" << endl;

        sum = 0;
    }

    return 0;
}